### PR TITLE
stageinfo,stage: implement stage hot-reloading

### PR DESCRIPTION
**Pull request by Akaricchi**
POSIX platforms only. Must be enabled via -Dstages_live_reload=true.

### CODE REVIEW

File: src/stageinfo.c [hunk 1]
```diff
@@ -104,37 +92,71 @@ static bool spellfilter_extra(AttackInfo *spell) {
...
+
 void stageinfo_init(void) {
+	dynstage_init();
+	stageinfo_fill(dynstage_get_exports());
+	stageinfo.stages_progress = calloc(stageinfo.stages.num_elements, sizeof(*stageinfo.stages_progress));
+}
+
+void stageinfo_reload(void) {
+	if(!dynstage_reload_library()) {
+		log_debug("dynstage not reloaded");
```
Comment (StarWitch) [inline src/stageinfo.c:105]: When it's turned _off_, this line can be a little spammy in the logs, mostly in the form of:

```
258894    D: stageinfo_reload: dynstage not reloaded
258894    D: stageinfo_reload: dynstage not reloaded
258894    D: stageinfo_reload: dynstage not reloaded
258894    D: stageinfo_reload: dynstage not reloaded
258894    D: stageinfo_reload: dynstage not reloaded
```
etc.

Minor nitpick, feel free to ignore.

Comment (Akaricchi) [reply src/stageinfo.c:105]: It's strange that this would be called multiple times in a row like that… worth investigating

Comment (StarWitch) [reply src/stageinfo.c:105]: IIRC, it was happening when I quit out of the main game. i.e:

1. Load Taisei, start main game
3. `Esc` -> `Stop the Game`
4. Log messages appeared

Comment (StarWitch) [reply src/stageinfo.c:105]: More context from logs. Seems to happen immediately after `progress_write`. And actually, additional log messages appear the first time (and only the first time) you play the game. Stopping the game and then going back to `Start Game` does not get you the additional `proc is missing` messages.

```
4940      D: progress_write: prepare 0: 8
4940      D: progress_write: prepare 1: 7
4940      D: progress_write: prepare 2: 9
4940      D: progress_write: prepare 3: 7
4940      D: progress_write: prepare 4: 58
4940      D: progress_write: prepare 5: 0
4940      D: progress_write: prepare 6: 6
4940      D: progress_write: prepare 7: 11
4940      D: progress_write: prepare 8: 11
4940      D: progress_write: prepare 9: 0
4940      D: progress_write: write 0: 8
4940      D: progress_write: write 1: 7
4940      D: progress_write: write 2: 9
4940      D: progress_write: write 3: 7
4940      D: progress_write: write 4: 58
4940      D: progress_write: write 5: 0
4940      D: progress_write: write 6: 6
4940      D: progress_write: write 7: 11
4940      D: progress_write: write 8: 11
4940      D: progress_write: write 9: 0
4940      D: stageinfo_reload: dynstage not reloaded
4940      D: _stage_enter: event proc is missing
4940      D: _stage_enter: update proc is missing
4940      D: stageinfo_reload: dynstage not reloaded
4940      D: _stage_enter: update proc is missing
4940      D: stageinfo_reload: dynstage not reloaded
4940      D: _stage_enter: update proc is missing
4940      D: stageinfo_reload: dynstage not reloaded
4940      D: _stage_enter: update proc is missing
4940      D: stageinfo_reload: dynstage not reloaded
4940      D: _stage_enter: update proc is missing
```

Comment (Akaricchi) [reply src/stageinfo.c:105]: I see. The issue here is that stage advancement in story mode is implemented in a kind of silly way: basically we loop over story stages and unconditionally call `stage_enter` on them, which exits early if `global.gameover` is set. Refactoring that mess is out of scope for this PR but I could at least move the exit condition before the log-spamming stuff is called.

**Pull request merged**